pkg/manifest/v1/objects.go

```diff
@@ -0,0 +1,156 @@
+const (
+	KindAlertCondition          = "AlertCondition"
+	KindAlertNotificationTarget = "AlertNotificationTarget"
+	KindAlertPolicy             = "AlertPolicy"
+	KindDataSource              = "DataSource"
+	KindSLI                     = "SLI"
+	KindSLO                     = "SLO"
+	KindService                 = "Service"
+)
+
+// Parse is responsible for parsing all structs in this apiVersion.
+func Parse(fileContent []byte, m ObjectGeneric, filename string) (manifest.OpenSLOKind, error) {
+	switch m.Kind {
+	case KindAlertCondition:
+		var content AlertCondition
+		err := yaml.Unmarshal(fileContent, &content)
+		return content, err
+	case KindAlertNotificationTarget:
+		var content AlertNotificationTarget
+		err := yaml.Unmarshal(fileContent, &content)
+		return content, err
+	case KindAlertPolicy:
+		var content AlertPolicy
+		err := yaml.Unmarshal(fileContent, &content)
+		return content, err
+	case KindDataSource:
+		var content DataSource
+		err := yaml.Unmarshal(fileContent, &content)
+		return content, err
+	case KindService:
+		var content Service
+		err := yaml.Unmarshal(fileContent, &content)
+		return content, err
+	case KindSLO:
+		var content SLO
+		err := yaml.Unmarshal(fileContent, &content)
+		return content, err
+	case KindSLI:
+		var content SLI
+		err := yaml.Unmarshal(fileContent, &content)
+		return content, err
+	default:
+		return nil, fmt.Errorf("unsupported kind: %s", m.Kind)
+	}
+}
+
+// ----------------------------------------------------------------------------
+// Object definitions
+// ----------------------------------------------------------------------------
+
+// Labels is a map of labels.
+type Labels map[string][]string
+
+// Annotations is a map of annotations.
+type Annotations map[string]string
+
+// Metadata represents part of object which is is common for all available Objects, for internal usage.
+type Metadata struct {
+	Name        string `yaml:"name" validate:"required" example:"name"`
+	DisplayName string `yaml:"displayName,omitempty" validate:"omitempty,min=0,max=63" example:"Prometheus Source"`
+	Labels      Labels `json:"labels,omitempty" validate:"omitempty,labels"`
+	Annotations Labels `json:"annotations,omitempty" validate:"omitempty,labels"`
+}
+
+// MetadataHolder is an intermediate structure that can provides metadata related
+// field to other structures.
+type MetadataHolder struct {
+	Metadata Metadata `yaml:"metadata"`
+}
+
+// ObjectHeader represents Header which is common for all available Objects.
+type ObjectHeader struct {
+	manifest.ObjectHeader `yaml:",inline"`
+	Kind                  string `yaml:"kind" validate:"required,oneof=Service SLO AlertNotificationTarget" example:"kind"`
+	MetadataHolder        `yaml:",inline"`
+}
+
+// ObjectGeneric represents struct to which every Objects is parsable
+// Specific types of Object have different structures as Spec.
+type ObjectGeneric struct {
+	ObjectHeader `yaml:",inline"`
+}
+
+// Indicator represents integration with metric source.
```

Comment:
For my understanding, what's the purpose of having these structs duplicating some of the other structs?

Reply:
Which ones are duplicated?

Reply:
Taking another look, is it that some alpha spec version structs are included in v1?

For example, `RatioMetrics` below is from alpha1, but the v1 struct is `RatioMetric` with addition of `Bad`. The same for `Indicator`.

Reply:
I see what you mean.  Yeah, theres `v1alpha` objects in the `v1/objects.go` file.  I'll remove

Reply:
Updated